backend/graph/api/collection-resolver.go

```diff
@@ -84,24 +106,42 @@ func sectionCollectionEntryResolver(
 		if err != nil {
 			return nil, err
 		}
-
-		limit := 20
-		if col.Filter != nil && col.Filter.Limit != nil {
-			limit = *col.Filter.Limit
+		entries = filterWithIds(col, entries, ids)
+	case "my_list":
+		ginCtx, err := utils.GinCtx(ctx)
+		if err != nil {
+			break
 		}
-		var newEntries []collection.Entry
-		for _, id := range utils.PointerIntArrayToIntArray(ids) {
-			entry, found := lo.Find(entries, func(e collection.Entry) bool {
-				return e.Collection == "episodes" && e.ID == strconv.Itoa(id)
-			})
-			if found {
-				newEntries = append(newEntries, entry)
-				if len(newEntries) >= limit {
-					break
-				}
-			}
+		profile := user.GetProfileFromCtx(ginCtx)
+		if profile == nil {
+			entries = []collection.Entry{}
+			break
+		}
+		myListID, err := appLoaders.UserMyListCollectionID.Get(ctx, profile.ID)
+		if err != nil || myListID == nil {
+			entries = []collection.Entry{}
+			break
 		}
-		entries = newEntries
+		entryIDs, err := ls.UserCollectionEntryIDsLoader.Get(ctx, *myListID)
+		if err != nil {
+			return nil, err
+		}
+		collectionEntries, err := ls.UserCollectionEntryLoader.GetMany(ctx, utils.PointerArrayToArray(entryIDs))
+		if err != nil {
+			return nil, err
+		}
+		// UUIDs, but only for episodes
+		uuids := lo.Map(lo.Filter(collectionEntries, func(i *common.UserCollectionEntry, _ int) bool {
+			return i.Type == "episode"
+		}), func(i *common.UserCollectionEntry, _ int) uuid.UUID {
+			return i.ItemID
+		})
+		ids, err := ls.EpisodeIDFromUuidLoader.GetMany(ctx, uuids)
+		if err != nil {
+			return nil, err
+		}
+
+		entries = filterWithIds(col, entries, ids)
```

Comment:
For consideration: Personally I feel that it is not very readable where this begins and ends. I would package the block (or major parts) as a function and call it in the case.

Reply:
Agreed.